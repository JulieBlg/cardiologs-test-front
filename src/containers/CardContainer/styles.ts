import { Theme } from '@material-ui/core';

const styles = (theme: Theme) => ({
    patientCardContainer: {
        padding: theme.spacing(3),
    }
})

export default styles;