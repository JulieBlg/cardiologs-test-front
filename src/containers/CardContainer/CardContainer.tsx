import React, { FC } from 'react';
import { Grid, Paper, Typography, makeStyles } from '@material-ui/core';

import PatientCard from '../../components/PatientCard';
import styles from './styles';
import { Card } from '../../typings';

interface Props {
    status: string;
    cards: Card[];
    handleClick: (card: Card, statusButton: string) => void;
}

const CardContainer: FC<Props> = ({ status, cards, handleClick }) => {
    const classes = makeStyles(styles)();

    return (
        <Grid item xs={4}>
            <Paper elevation={2} className={classes.patientCardContainer}>
                <Typography variant="h5" component="h2" align="center">
                    {status}
                </Typography>
                {cards.map(card => {
                    return <PatientCard key={card.id} card={card} status={status} handleClick={handleClick}/>
                })}
            </Paper>
        </Grid>
    );
}

export default CardContainer;