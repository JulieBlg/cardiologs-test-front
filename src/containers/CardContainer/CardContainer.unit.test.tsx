import React from 'react';
import { render } from '@testing-library/react';

import CardContainer from './CardContainer';

const mockCards = [
    {
        "arrhythmias": [
            "AFib",
            "AV Block",
            "Pause",
            "PSVC",
            "PVC"
        ],
        "created_date": "2020-03-10T13:14:59+0000",
        "id": 0,
        "patient_name": "Bob",
        "status": "PENDING"
    },
    {
        "arrhythmias": [
            "Pause"
        ],
        "created_date": "2020-01-01T00:12:21+0000",
        "id": 1,
        "patient_name": "Bill",
        "status": "PENDING"
    },
    {
        "arrhythmias": [
            "AFib",
            "Pause"
        ],
        "created_date": "2019-12-31T00:11:14+0000",
        "id": 2,
        "patient_name": "Elsa",
        "status": "PENDING"
    },
];

describe('CardContainer', () => {
  it('should render CardContainer properly', () => {
    const { container } = render(<CardContainer status="PENDING" cards={mockCards} handleClick={jest.fn()}/>);

    expect(container).toMatchSnapshot();
  });
})

