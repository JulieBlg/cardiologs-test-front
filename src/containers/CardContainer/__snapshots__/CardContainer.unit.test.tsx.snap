// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should render CardContainer properly 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
  >
    <div
      class="MuiPaper-root makeStyles-patientCardContainer-1 MuiPaper-elevation2 MuiPaper-rounded"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
      >
        PENDING
      </h2>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-patientCard-163 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiCardHeader-root"
            >
              <div
                class="MuiCardHeader-avatar"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
                >
                  B
                </div>
              </div>
              <div
                class="MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
                >
                  Bob
                </span>
                <span
                  class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                >
                  Patient ID : 0
                </span>
              </div>
            </div>
            <div
              class="MuiCardContent-root"
            >
              <p
                class="MuiTypography-root makeStyles-listTitle-164 MuiTypography-body1"
              >
                Arrhythmias :
              </p>
              <div
                class="MuiGrid-root makeStyles-arrythmiasList-165 MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiChip-root makeStyles-arrythmia-166"
                >
                  <span
                    class="MuiChip-label"
                  >
                    AFib
                  </span>
                </div>
                <div
                  class="MuiChip-root makeStyles-arrythmia-166"
                >
                  <span
                    class="MuiChip-label"
                  >
                    AV Block
                  </span>
                </div>
                <div
                  class="MuiChip-root makeStyles-arrythmia-166"
                >
                  <span
                    class="MuiChip-label"
                  >
                    Pause
                  </span>
                </div>
                <div
                  class="MuiChip-root makeStyles-arrythmia-166"
                >
                  <span
                    class="MuiChip-label"
                  >
                    PSVC
                  </span>
                </div>
                <div
                  class="MuiChip-root makeStyles-arrythmia-166"
                >
                  <span
                    class="MuiChip-label"
                  >
                    PVC
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiCardActions-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-actionButton-167 MuiButton-containedSizeSmall MuiButton-sizeSmall"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      COMPLETED
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-actionButton-167 MuiButton-containedSizeSmall MuiButton-sizeSmall"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      DONE
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-patientCard-247 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiCardHeader-root"
            >
              <div
                class="MuiCardHeader-avatar"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
                >
                  B
                </div>
              </div>
              <div
                class="MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
                >
                  Bill
                </span>
                <span
                  class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                >
                  Patient ID : 1
                </span>
              </div>
            </div>
            <div
              class="MuiCardContent-root"
            >
              <p
                class="MuiTypography-root makeStyles-listTitle-248 MuiTypography-body1"
              >
                Arrhythmias :
              </p>
              <div
                class="MuiGrid-root makeStyles-arrythmiasList-249 MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiChip-root makeStyles-arrythmia-250"
                >
                  <span
                    class="MuiChip-label"
                  >
                    Pause
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiCardActions-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-actionButton-251 MuiButton-containedSizeSmall MuiButton-sizeSmall"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      COMPLETED
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-actionButton-251 MuiButton-containedSizeSmall MuiButton-sizeSmall"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      DONE
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-patientCard-252 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiCardHeader-root"
            >
              <div
                class="MuiCardHeader-avatar"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
                >
                  E
                </div>
              </div>
              <div
                class="MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
                >
                  Elsa
                </span>
                <span
                  class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                >
                  Patient ID : 2
                </span>
              </div>
            </div>
            <div
              class="MuiCardContent-root"
            >
              <p
                class="MuiTypography-root makeStyles-listTitle-253 MuiTypography-body1"
              >
                Arrhythmias :
              </p>
              <div
                class="MuiGrid-root makeStyles-arrythmiasList-254 MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiChip-root makeStyles-arrythmia-255"
                >
                  <span
                    class="MuiChip-label"
                  >
                    AFib
                  </span>
                </div>
                <div
                  class="MuiChip-root makeStyles-arrythmia-255"
                >
                  <span
                    class="MuiChip-label"
                  >
                    Pause
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiCardActions-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-actionButton-256 MuiButton-containedSizeSmall MuiButton-sizeSmall"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      COMPLETED
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-actionButton-256 MuiButton-containedSizeSmall MuiButton-sizeSmall"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      DONE
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
